include/chemfiles/formats/PDB.hpp

```diff
@@ -46,10 +46,12 @@ class PDBFormat final: public Format {
                         const std::string& sec);
     // Read CONECT record
     void read_CONECT(Frame& frame, const std::string& line);
+    // Runs when a chain is terminated to update residue information
+    void chain_ended(Frame& frame);
 
     std::unique_ptr<TextFile> file_;
-    /// Map of residues, indexed by residue id and chainid.
-    std::map<std::pair<char, size_t>, Residue> residues_;
+    /// Map of residues, indexed by residue id, chain id and insertion code.
+    std::map<std::tuple<char, size_t, char>, Residue> residues_;
```

Comment:
The travis failure might be related to std::tuple not having std::hash implementation. Maybe you can replace the tuple by a small struct `residue_index` ?

Reply:
It's a C++ version thing (I use 14 by default by accident). I like using a tuple because I get the comparison operator for free. I may need to write my own anyway to fix how secondary structures are parsed as the use of `' '` is a hack.